plugins/services/src/js/components/modals/NewCreateServiceModal.js

```diff
@@ -371,16 +384,28 @@ class NewServiceFormModal extends Component {
     // Switch directly to form if edit
     if (nextProps.isEdit) {
       newState.servicePickerActive = false;
-      newState.serviceFormActive = true;
+
+      if (nextProps.service instanceof Pod) {
+        newState.serviceJsonActive = true;
+      } else {
+        newState.serviceFormActive = true;
+      }
     }
 
     return newState;
   }
 
   getSecondaryActions() {
-    let {servicePickerActive, serviceFormActive} = this.state;
+    let {
+      serviceFormActive,
+      servicePickerActive,
+      serviceReviewActive,
+      serviceJsonActive
+    } = this.state;
     let label = 'Back';
-    if (servicePickerActive || (serviceFormActive && this.props.isEdit)) {
+    if (servicePickerActive ||
```

Comment:
Let's also add a newline above here - this block of `let`s and `if` is hard to parse.